snapcraft/plugins/catkin.py

```diff
@@ -889,9 +889,11 @@ def find(self, package_name):
 
     def _run(self, arguments):
         with tempfile.NamedTemporaryFile(mode='w+') as f:
-            lines = ['export PYTHONPATH={}'.format(os.path.join(
-                self._catkin_install_path, 'usr', 'lib', 'python2.7',
-                'dist-packages'))]
+            lines = ['export PYTHONPATH={}:{}'.format(
+                os.path.join(self._catkin_install_path, 'usr', 'lib',
+                             'python2.7', 'dist-packages'),
+                os.path.join(self._catkin_install_path, 'opt', 'ros',
+                             'kinetic', 'lib', 'python2.7', 'dist-packages'))]
```

Comment:
This is odd: this change solves your issue? I need to do some testing to be sure, but I swear that sourcing the `setup.sh` as done immediately below this adds that to the PYTHONPATH.

Note also that this hard-codes `kinetic`, it should be using the `ros_path` defined below instead.